src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -0,0 +1,333 @@
+                {
+                    var argumentOpt = TryGetRelevantArgument(initialNode, node, diagnostic);
+                    await HandleObjectCreationExpressionAsync(context, objectCreation, argumentOpt).ConfigureAwait(false);
+                    return;
+                }
+                else if (node is TInvocationExpressionSyntax invocationExpression)
+                {
+                    var argumentOpt = TryGetRelevantArgument(initialNode, node, diagnostic);
+                    await HandleInvocationExpressionAsync(context, invocationExpression, argumentOpt).ConfigureAwait(false);
+                    return;
+                }
+            }
+        }
+
+        private TArgumentSyntax TryGetRelevantArgument(
+            SyntaxNode initialNode, SyntaxNode node, Diagnostic diagnostic)
+        {
+            if (this.TooManyArgumentsDiagnosticIds.Contains(diagnostic.Id))
+            {
+                return null;
+            }
+
+            return initialNode.GetAncestorsOrThis<TArgumentSyntax>()
+                              .LastOrDefault(a => a.AncestorsAndSelf().Contains(node));
+        }
+
+        private Task HandleInvocationExpressionAsync(
+            CodeFixContext context, TInvocationExpressionSyntax invocationExpression, TArgumentSyntax argumentOpt)
+        {
+            // Currently we only support this for 'new obj' calls.
+            return SpecializedTasks.EmptyTask;
+        }
+
+        private async Task HandleObjectCreationExpressionAsync(
+            CodeFixContext context,
+            TObjectCreationExpressionSyntax objectCreation,
+            TArgumentSyntax argumentOpt)
+        {
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+
+            // Not supported if this is "new Foo { ... }" (as there are no parameters at all.
+            var typeNode = syntaxFacts.GetObjectCreationType(objectCreation);
+            if (typeNode == null)
+            {
+                return;
+            }
+
+            // If we can't figure out the type being created, or the type isn't in source,
+            // then there's nothing we can do.
+            var type = semanticModel.GetSymbolInfo(typeNode, cancellationToken).GetAnySymbol() as INamedTypeSymbol;
+            if (type == null)
+            {
+                return;
+            }
+
+            if (!type.IsNonImplicitAndFromSource())
+            {
+                return;
+            }
+
+            var arguments = (SeparatedSyntaxList<TArgumentSyntax>)syntaxFacts.GetArgumentsOfObjectCreationExpression(objectCreation);
+
+            var comparer = syntaxFacts.IsCaseSensitive
+                ? StringComparer.Ordinal
```

Comment:
All the checks are done to the 'interpreted' name.  so all escapes are gone at that point.